vsintegration/src/FSharp.Editor/LanguageService/FSharpProjectOptionsManager.fs

```diff
@@ -56,32 +56,56 @@ module private FSharpProjectOptionsHelpers =
+        let oldProjectMetadataRefs = oldProject.MetadataReferences
+        let newProjectMetadataRefs = newProject.MetadataReferences
+
+        if oldProjectMetadataRefs.Count <> newProjectMetadataRefs.Count then true
+        else
+
+        let mutable mustCheckFcsInvalidation = false
+
         let oldProjectRefs = oldProject.ProjectReferences
         let newProjectRefs = newProject.ProjectReferences
-        oldProjectRefs.Count() <> newProjectRefs.Count() ||
-        (oldProjectRefs, newProjectRefs)
-        ||> Seq.exists2 (fun p1 p2 ->
-            ct.ThrowIfCancellationRequested()
-            let doesProjectIdDiffer = p1.ProjectId <> p2.ProjectId
-            let p1 = oldProject.Solution.GetProject(p1.ProjectId)
-            let p2 = newProject.Solution.GetProject(p2.ProjectId)
-            doesProjectIdDiffer || 
-            (
-                // For F# projects, just check the version until we have a better in-memory model for them.
-                if p1.Language = LanguageNames.FSharp then
-                    p1.Version <> p2.Version
-                else
-                    let v1 = p1.GetDependentVersionAsync(ct).Result
-                    let v2 = p2.GetDependentVersionAsync(ct).Result
-                    v1 <> v2
+
+        let res =
+            oldProjectRefs.Count() <> newProjectRefs.Count() ||
+            (oldProjectRefs, newProjectRefs)
+            ||> Seq.exists2 (fun p1 p2 ->
+                ct.ThrowIfCancellationRequested()
+                let doesProjectIdDiffer = p1.ProjectId <> p2.ProjectId
+                let p1 = oldProject.Solution.GetProject(p1.ProjectId)
+                let p2 = newProject.Solution.GetProject(p2.ProjectId)
+                doesProjectIdDiffer || 
+                (
+                    if p1.Language = LanguageNames.FSharp then
+                        if p1.Version <> p2.Version then
+                            true
+                        else
+                            let v1 = p1.GetDependentVersionAsync(ct).Result
+                            let v2 = p2.GetDependentVersionAsync(ct).Result
+                            if v1 <> v2 then
+                                mustCheckFcsInvalidation <- true
+                            false
+                    else
+                        let v1 = p1.GetDependentVersionAsync(ct).Result
+                        let v2 = p2.GetDependentVersionAsync(ct).Result
+                        v1 <> v2
+                )
             )
-        )
 
-    let isProjectInvalidated (oldProject: Project) (newProject: Project) (settings: EditorOptions) ct =
+        if not res && mustCheckFcsInvalidation then
+            // At the moment, Roslyn's view of F# dependent project references must not cause an invalidation
+            // because the internals of FCS's state relies on the file-system.
+            // Therefore, check if the project is invalidated by FCS's view of the world.
+            checker.IsProjectReferencesInvalidated options
```

Comment:
This isn't necessary. FCS will check the invalidation when trying to get an internal incremental build; if it's invalidated, it will just re-create it.